src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/MarkDeliveredModal.tsx

```diff
@@ -0,0 +1,124 @@
+import { Controller, useForm } from 'react-hook-form';
+import NERFormModal from '../../../components/NERFormModal';
+import { useMarkReimbursementRequestAsDelivered } from '../../../hooks/finance.hooks';
+import { useToast } from '../../../hooks/toasts.hooks';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';
+import { DatePicker } from '@mui/x-date-pickers';
+import { ReimbursementRequest, startOfDay } from 'shared';
+
+const schema = yup.object().shape({
+  dateDelivered: yup.date().required('Must provide delivery date.'),
+  confirmDelivered: yup
+    .boolean()
+    .required('Please confirm items delivered.')
+    .test('is-true', 'Please confirm', (value) => value === true)
+});
+
+interface MarkDeliveredModalProps {
+  modalShow: boolean;
+  onHide: () => void;
+  reimbursementRequest: ReimbursementRequest;
+}
+
+const MarkDeliveredModal = ({ modalShow, onHide, reimbursementRequest }: MarkDeliveredModalProps) => {
+  const toast = useToast();
+  const { mutateAsync: markDelivered } = useMarkReimbursementRequestAsDelivered(reimbursementRequest.reimbursementRequestId);
+
+  const dateIsBeforeExpenseCreated = (date: Date): boolean => {
```

Comment:
Could you rename this to `dateIsBeforeExpense` - we want to make sure we're clear this isn't when the expense was reported